torch_geometric/nn/aggr/basic.py

```diff
@@ -160,9 +164,14 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
                 ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
                 dim: int = -2) -> Tensor:
 
+        if self.channels > 1:
```

Comment:
This does not seem like a requirement.

Reply:
You mean we should always check for two dimensional input, or we should never? 

I added this because in the case of a per channel `t` we need to infer which dimension in the input is the channel dimension. I thought supporting two dimensions only was simpler than taking also a channel dim input.

WDYT?

Reply:
Ah, you are right, thanks for clarifying.